pkg/mapper/iac-providers/cft/config/iam-role.go

```diff
@@ -0,0 +1,68 @@
+package config
+
+import (
+	"encoding/json"
+
+	"github.com/awslabs/goformation/v4/cloudformation/iam"
+)
+
+const (
+	// IamRolePolicy represents subresource aws_iam_role_policy for attribute Policy
+	IamRolePolicy = "Policy"
+)
+
+// IamRoleConfig holds config for aws_iam_role
+type IamRoleConfig struct {
+	Config
+	RoleName                 string `json:"name"`
+	AssumeRolePolicyDocument string `json:"assume_role_policy"`
+}
+
+// IamRolePolicyConfig holds config for aws_iam_role_policy
+type IamRolePolicyConfig struct {
+	Config
+	PolicyName     string `json:"name"`
+	PolicyDocument string `json:"policy"`
+}
+
+// GetIamRoleConfig returns config for aws_iam_role and aws_iam_role_policy
+func GetIamRoleConfig(r *iam.Role) []AWSResourceConfig {
+	resourceConfigs := make([]AWSResourceConfig, 0)
+
+	// add aws_iam_role
+	roleConfig := IamRoleConfig{
+		Config: Config{
+			Name: r.RoleName,
+			Tags: r.Tags,
+		},
+	}
+	policyDocument, err := json.Marshal(r.AssumeRolePolicyDocument)
+	if err == nil {
+		roleConfig.AssumeRolePolicyDocument = string(policyDocument)
+	}
+	resourceConfigs = append(resourceConfigs, AWSResourceConfig{Resource: roleConfig})
+
+	// aws_iam_role_policy as a SubResource
+	// multiple Policies can be defined for a resource in cft
+	if r.Policies != nil {
```

Comment:
is it required to create a separate resource for policies when the policy could be part of iam role properties?

Reply:
It is required because Policy is just a property in CloudFormation but a resource in terraform. There are many other resources where this is the case.

Reply:
Thanks @sigmabaryon. In that case should it also be part of the respective cft resource properties?